src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -57,6 +57,18 @@ function getUnmaskedContext(workInProgress : Fiber) : Object {
 }
 exports.getUnmaskedContext = getUnmaskedContext;
 
+function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
+  if (isContextConsumer(workInProgress)) {
```

Comment:
You are correct. We do not need `instance` for that. I just mentioned that we need to do _some_ kind of duplicate work since `instance` won't always exist.

`getMaskedContext` returns `emptyObject` for non-consumers. We could compare it? eg

```js
function cacheContext(workInProgress : Fiber, unmaskedContext : Object, maskedContext : Object) {
  if (maskedContext !== emptyObject) {
    const instance = workInProgress.stateNode;

    if (instance) {
      instance.__reactInternalMemoizedUnmaskedChildContext = unmaskedContext;
      instance.__reactInternalMemoizedMaskedChildContext = maskedContext;
    }
  }
}
```